### PR TITLE
fix: ignore fork branch when get_cellbase_output_capacity_details, get_header and get_block rpc

### PR DESCRIPTION
**Pull request by u2**

### CODE REVIEW

File: rpc/src/module/chain.rs [hunk 1]
```diff
@@ -81,11 +82,13 @@ impl ChainRpc for ChainRpcImpl {
...
     fn get_header(&self, hash: H256) -> Result<Option<HeaderView>> {
-        Ok(self
-            .shared
-            .snapshot()
-            .get_block_header(&hash.pack())
-            .map(Into::into))
+        let snapshot = self.shared.snapshot();
+
+        if !snapshot.is_main_chain(&hash.pack()) {
+            return Ok(None);
```
Comment (keroro520) [inline rpc/src/module/chain.rs]: What if I wanna get an uncle-block? Add another RPC interface later?

Comment (u2) [reply rpc/src/module/chain.rs]: fork-block? `get_fork_block` is only for the test now.

Comment (keroro520) [reply rpc/src/module/chain.rs]: Yeah, that is what I want. Thanks.

**Comment by zhangsoledad** [comment]
bors r=keroro520,quake,zhangsoledad

**Comment by u2** [comment]
Please rerun `bors r+`, there is an error, but I think it's not introduced by this PR.

https://travis-ci.com/nervosnetwork/ckb/jobs/248970692
```
[2019-10-24T02:31:56Z INFO  ckb_test] 45/88 .............. Running compact_block_missing_fresh_txs
[2019-10-24T02:31:56Z INFO  ckb_test] Started Net with working dir: /var/folders/nz/vv4_9tw56nv9k3tkvyszvwg80000gn/T/.tmpRELDlj
[2019-10-24T02:31:59Z INFO  ckb_network::config] Generate random key
[2019-10-24T02:31:59Z INFO  ckb_network::config] write random secret key to "/var/folders/nz/vv4_9tw56nv9k3tkvyszvwg80000gn/T/.tmpRELDlj/secret_key"
[2019-10-24T02:31:59Z INFO  ckb_network::network] Listen on address: /ip4/127.0.0.1/tcp/9000/p2p/QmNnFT3GgkBvoJ8Sev5urmDJTKrHJxvN5rNYWdagsG7xmv
[2019-10-24T02:31:59Z INFO  ckb_network::network] p2p service event: ListenStarted { address: "/ip4/127.0.0.1/tcp/9000" }
thread 'main' panicked at 'Node0 should send GetBlockTransactions message for missing transactions', src/net.rs:234:9
```

**Comment by u2** [comment]
> bors r+

**Comment by u2** [comment]
Please `rerun bors r+`, @keroro520, thanks.

**Pull request merged**